subprojects/architecture-test/src/test/java/org/gradle/architecture/test/DependencyInjectionTest.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.architecture.test;
+
+import com.tngtech.archunit.base.DescribedPredicate;
+import com.tngtech.archunit.core.domain.JavaClass;
+import com.tngtech.archunit.junit.AnalyzeClasses;
+import com.tngtech.archunit.junit.ArchTest;
+import com.tngtech.archunit.lang.ArchRule;
+import org.gradle.internal.service.scopes.ServiceScope;
+
+import javax.inject.Inject;
+
+import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.classes;
+import static org.gradle.architecture.test.ArchUnitFixture.freeze;
+
+@AnalyzeClasses(packages = "org.gradle")
+public class DependencyInjectionTest {
```

Comment:
🤔 This name is not very descriptive.  Maybe `ScopeAnnotationValidationTest`?  

Also, some short javadoc explaining the problem and goal would be good.  Maybe along with a link to a bt-internal-docs page?

Reply:
The naming, I agree with!

About creating a page on bt-internal: I think it's much better if I create the JavaDoc here. The tests are really easy and fast to reproduce (just run `sC`), and after double-clicking on the test, the developer will be navigated here.

Reply:
Renamed!